packages/contracts-bedrock/tasks/genesis.ts

```diff
@@ -0,0 +1,268 @@
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+    } = require('@defi-wonderland/smock/dist/src/utils')
+
+    // Deploying to an already existing network, the existing address
+    // should be used
+    let ProxyL1CrossDomainMessenger = await hre.deployments.getOrNull(
+      'Proxy__OVM_L1CrossDomainMessenger'
+    )
+    // When deploying to a fresh network, use the freshly deployed
+    // contract. TODO: this will be placed behind a proxy eventually
+    // and need to be updated
+    if (ProxyL1CrossDomainMessenger === undefined) {
+      ProxyL1CrossDomainMessenger = await hre.deployments.get(
+        'L1CrossDomainMessenger'
+      )
+    }
+
+    let ProxyL1StandardBridge = await hre.deployments.getOrNull(
+      'Proxy__OVM_L1StandardBridge'
+    )
+    if (ProxyL1StandardBridge === undefined) {
+      // TODO: update to use proxy addr
+      ProxyL1StandardBridge = await hre.deployments.get('L1StandardBridge')
+    }
+
+    const variables = {
+      L2ToL1MessagePasser: {
+        nonce: 0,
+      },
+      L2CrossDomainMessenger: {
+        _initialized: 1,
+        _owner: hre.deployConfig.l2CrossDomainMessengerOwner,
+        xDomainMsgSender: '0x000000000000000000000000000000000000dEaD',
+        msgNonce: 0,
+        otherMessenger: ProxyL1CrossDomainMessenger.address,
+        // TODO: handle blockedSystemAddresses mapping
+        // blockedSystemAddresses: [{key: '', value: ''}],
+      },
+      GasPriceOracle: {
+        _owner: hre.deployConfig.gasPriceOracleOwner,
+        overhead: hre.deployConfig.gasPriceOracleOverhead,
+        scalar: hre.deployConfig.gasPriceOracleScalar,
+        decimals: hre.deployConfig.gasPriceOracleDecimals,
+      },
+      L2StandardBridge: {
+        messenger: predeploys.L2CrossDomainMessenger,
+        otherBridge: ProxyL1StandardBridge.address,
+      },
+      SequencerFeeVault: {
+        l1FeeWallet: ethers.constants.AddressZero,
+      },
+      OptimismMintableTokenFactory: {
+        bridge: ethers.constants.AddressZero,
+      },
+      L1Block: {
+        number: hre.deployConfig.l1BlockInitialNumber,
+        timestamp: hre.deployConfig.l1BlockInitialTimestamp,
+        basefee: hre.deployConfig.l1BlockInitialBasefee,
+        hash: hre.deployConfig.l1BlockInitialHash,
+        sequenceNumber: hre.deployConfig.l1BlockInitialSequenceNumber,
+      },
+      OVM_ETH: {
+        bridge: predeploys.L2StandardBridge,
+        remoteToken: ethers.constants.AddressZero,
+        _name: 'Ether',
+        _symbol: 'ETH',
+      },
+      WETH9: {
+        name: 'Wrapped Ether',
+        symbol: 'WETH',
+        decimals: 18,
+      },
+      GovernanceToken: {
+        name: 'Optimism',
+        symbol: 'OP',
+        _owner: hre.deployConfig.proxyAdmin,
+      },
+    }
+
+    assertEvenLength(implementationSlot)
+    assertEvenLength(adminSlot)
+    assertEvenLength(hre.deployConfig.proxyAdmin)
+
+    const predeployAddrs = new Set(Object.values(predeploys))
```

Comment:
This set should be initialized as empty and all address comparisons should be done on checksummed addresses